README.md

```diff
@@ -1,16 +1,1831 @@
+  fork      [&Invocation]
+  
+  # Additional task to be invoked with added semantics
+  # of representing a workflow execution continuation.
+  join       optional &Invocation
+}
+```
+
+## 7.2 Fields
+
+### 7.2.1 Forked Task Invocations
+
+The OPTIONAL `fork` field, if present MUST be a list of an alphabetically ordered [Invocation] links. List MUST NOT not contain duplicate entries. Every linked [Invocation] MUST have `cause` field set to the same [Invocation] (link) as the `run` field of the containing [Receipt].
+
+### 7.2.2 Joined Task Invocation
+
+The OPTIONAL `join` field, if present MUST be set to an [Invocation] link. Linked [Invocation] in the `join` field MUST have `cause` field set to the same [Invocation] (link) as the `run` field of the containing [Receipt].
+
+## 7.4 DAG-JSON Examples
+
+### 7.4.1 Effect spawning concurrent threads
+
+```json
+{
+  "fork": [
+    {
+      "/": "bafyreigtbbrfic7gduybn2lcncbbzycbef3kt3aqeguko5dmuhpmf73zmu"
+    },
+    {
+      "/": "bafyreidj2fkidh5g4tlldael7dfw27va6vbn6ah6pv7hirsandzwp25pj4"
+    }
+  ]
+}
+```
+
+### 7.4.2 Effect continuing thread execution
+
+```json
+{
+  "join": {
+    "/": "bafyreigqsmtpw6qliojfzupzbxsnkn4yave7vov2jvvrtye7w4nomvd7yq"
+  }
+}
+```
+
+### 7.4.1 Effect with fork & join
+
+```json
+{
+  "fork": [
+    {
+      "/": "bafyreieljernknkcrmefokxxperobictfndo4v4cazkjmqfwuog4puawxm"
+    },
+    {
+      "/": "bafyreiho3x2lbi6a5irw3orgrzrivkakr7iymalpukmxsvdsoqlxpcjsfq"
+    }
+  ],
+  "join": {
+    "/": "bafyreidtsxqd5fcuwdgverklwpcgbgzixsf6z6wgelybiakpwec3q75iwa"
+  }
+}
+```
+
+# 8 Receipt
+
+A `Receipt` is an attestation of the [Result] and a caused [Effect] by the [Task] [Invocation]. A Receipt MUST be signed by the [Executor] or it's delegate. If signed by the delegate, the proof of delegation from the [Executor] to the Issuer (the `iss` of the receipt) MUST be provided in `prf`.
+
+**NB: a Receipt this does not guarantee correctness of the result!** The statement's veracity MUST be only understood as an attestation from the executor.
+
+Receipts MUST use the same version as the invocation that they contain.
+
+## 8.1 Schema
+
+```ipldsch
+type Receipt struct {
+  ran     &Invocation
+
+  # output of the invocation
+  out     Result
+  # Effects to be performed
+  fx      Effect
+
+  # All the other metadata
+  meta    {String: any}
+
+  # Principal that issued this receipt. If omitted issuer is
+  # inferred from the invocation task audience.
+  iss     optional Principal
+
+  # When issuer is different from executor this MUST hold a UCAN
+  # delegation chain from executor to the issuer. Should be omitted executor is an issuer.
+  prf     [&UCAN]
+
+  # Signature from the `iss`.
+  s       Varsig
+}
+```
+
+## 8.2 Fields
+
+### 8.2.1 Ran Invocation
+
+The `ran` field MUST include a link to the [Invocation] that the Receipt is for.
+
+### 8.2.2 Output
+
+The `out` field MUST contain the output of the invocation in [Result] format.
```

Comment:
Since we've added `fx`, how about something like:

```suggestion
The `out` field MUST contain the value output of the invocation in [Result] format.
```